wrappers/src/marshalling.hpp

```diff
@@ -105,6 +105,29 @@ typedef struct realm_value {
     }
 } realm_value_t;
 
+typedef struct realm_sync_error_compensating_write_info {
```

Comment:
The `typedef struct foobar { ... } foobar` syntax is unnecessary in C++.
Simply declaring
```c++
struct foobar {
   ...
};
```
is sufficient. `realm_value_t` was copied from a C header, where the declaration needs to be more verbose.